vision/textDetection.js

```diff
@@ -0,0 +1,284 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START app]
+// [START import_libraries]
+var async = require('async');
+var fs = require('fs');
+var path = require('path');
+var gcloud = require('gcloud');
+var natural = require('natural');
+var redis = require('redis');
+// [END import_libraries]
+
+// [START authenticate]
+// You must set the GOOGLE_APPLICATION_CREDENTIALS and GCLOUD_PROJECT
+// environment variables to run this sample. See:
+// https://github.com/GoogleCloudPlatform/gcloud-node/blob/master/docs/authentication.md
+var projectId = process.env.GCLOUD_PROJECT;
```

Comment:
You shouldn't have to do this really.  If you set this env var, gcloud just figures it out.  You should be able to drop this line, and the projectId config below.